src/main/java/org/dependencytrack/policy/cel/CelPolicyQueryManager.java

```diff
@@ -612,6 +606,26 @@ List<UUID> getParents(final UUID uuid, final List<UUID> parents) {
         return getParents(parentUuid, parents);
     }
 
+    boolean isDirectDependency(final org.dependencytrack.proto.policy.v1.Component component) {
+        //Using _ in the first part instead of : as persistence manager not returning result with : even with match
+        String likePart1 = """
+                '%"uuid"_""";
+        String likePart2 = """
+                "%s"%%'""".formatted(component.getUuid());
+        String query1 = """
+                SELECT COUNT(*) FROM "COMPONENT" "C" INNER JOIN "PROJECT" "P" ON "P"."ID"="C"."PROJECT_ID" WHERE "C"."UUID"='%s' AND "P"."DIRECT_DEPENDENCIES" LIKE
+                """.formatted(component.getUuid());
+        final Query<?> query = pm.newQuery(Query.SQL, query1 + likePart1 + likePart2);
```

Comment:
This is a bit dangerous as it could make the query vulnerable to SQL injection (assuming `component` is provided by the user).

I haven't tested it yet, but it should work with query parameters as well, e.g.:

```java
final var queryStr = """
    SELECT COUNT(*)
    FROM "COMPONENT" "C" 
    INNER JOIN "PROJECT" "P" ON "P"."ID" = "C"."PROJECT_ID"
    WHERE "C"."UUID" = :uuid
        AND "P"."DIRECT_DEPENDENCIES" LIKE :wildcard
    """;
final Query<?> query = pm.newQuery(Query.SQL, queryStr);
query.setNamedProperties(Map.of(
    "uuid", component.getUuid(),
    "wildcard", "%" + component.getUuid() + "%"
```